main/coursecopy/export_moodle.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/* For licensing terms, see /license.txt */
+
+use Chamilo\CourseBundle\Component\CourseCopy\CourseBuilder;
+use Chamilo\CourseBundle\Component\CourseCopy\CourseSelectForm;
+use moodleexport\MoodleExport;
+
+/**
+ * Create a Moodle export.
+ *
+ */
+require_once __DIR__.'/../inc/global.inc.php';
+require_once api_get_path(SYS_PATH).'main/work/work.lib.php';
+
+$current_course_tool = TOOL_COURSE_MAINTENANCE;
+
+api_protect_course_script(true);
+
+// Check access rights (only teachers are allowed here)
+if (!api_is_allowed_to_edit()) {
+    api_not_allowed(true);
+}
+
+api_check_archive_dir();
+api_set_more_memory_and_time_limits();
+
+// Section for the tabs
+$this_section = SECTION_COURSES;
+
+// Breadcrumbs
+$interbreadcrumb[] = [
+    'url' => api_get_path(WEB_CODE_PATH).'course_info/maintenance.php',
+    'name' => get_lang('Maintenance'),
+];
+
+// Displaying the header
+$nameTools = get_lang('ExportToMoodle');
+Display::display_header($nameTools);
+
+// Display the tool title
+echo Display::page_header($nameTools);
+$action = isset($_POST['action']) ? $_POST['action'] : '';
+$exportOption = isset($_POST['export_option']) ? $_POST['export_option'] : '';
+
+// Handle course selection form submission
+if ($action === 'course_select_form' && Security::check_token('post')) {
+    // Handle the selected resources and continue with export
+    $selectedResources = $_POST['resource'] ?? null;
+
+    if (!empty($selectedResources)) {
+        // Rebuild the course object based on selected resources
+        $cb = new CourseBuilder('partial');
+        $course = $cb->build(0, null, false, array_keys($selectedResources), $selectedResources);
+
+        // Get admin details
+        $adminId = (int) $_POST['admin_id'];
+        $adminUsername = $_POST['admin_username'];
+        $adminEmail = $_POST['admin_email'];
+
+        $exporter = new MoodleExport($course);
+        $exporter->setAdminUserData($adminId, $adminUsername, $adminEmail);
+
+        // Perform export
+        $courseId = api_get_course_id();
+        $exportDir = 'moodle_export_' . $courseId;
+        try {
+            $moodleVersion = isset($_POST['moodle_version']) ? $_POST['moodle_version'] : '3';
```

Comment:
If we known $moodleVersion must be an integer value, we must ensure $_POST['moodle_version'] is an int (cast as (int), for example)